driver-core/src/main/java/com/datastax/driver/core/QueryLogger.java

```diff
@@ -551,16 +552,17 @@ private int append(Statement statement, StringBuilder buffer, int remaining) {
     }
 
     private int append(CharSequence str, StringBuilder buffer, int remaining) {
-        if (remaining == -1) {
+        if (remaining == -2) {
```

Comment:
I discovered a minor bug where `TRUNCATED_OUTPUT` wouldn't be printed if the last inserted string fits exactly in the buffer's remaining capacity.